### ISSUE TITLE
pyficache rationale?

### ISSUE THREAD

**Issue by NouberNou** [doorstop-dev/doorstop#421]
What is the rationale behind using pyficache? I experienced some weird issues with it not being able to get files during ref searches and I dug into it a bit. It seems way more complicated than what is needed for Doorstop ref searching.

I hacked in something in the meantime that is a bit simpler (though seems to meet the general requirements of what pyficache was achieving) and it seemed to have solved my issues, but I was wondering if I am missing something.

**Comment by jacebrowning** [comment]
The rationale was to avoid reading any file more than once while searching for references.

Can you share your alternative?

**Comment by NouberNou** [comment]
Currently my alternative is a pretty simple named hash cache where the path of the file is used as a key to an array representing the lines of code. It's essentially the same thing, but pyficache was giving me inconsistent results when I deployed it on Linux for some reason (worked fine on Windows). I removed some of complexity.

I need to formalize my implementation a bit more, it was sort of a POC/test that's made it further into my use than I'd planned. I'd like to add memory constraint options (total memory consumed by the cache, max/min file sizes to cache, possibly filtering on binary/text content).

I'll try to get a PR made up in the next few weeks for review.
